### PR TITLE
Parse URNs in batches

### PR DESCRIPTION
**Pull request by courte**
This branch supports an attempt to limit the number of repetitive database touches in the Metadata Wrangler by parsing URNs in batches. It also allows identifiers to be added to a collections catalog in batches.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -1519,9 +1522,8 @@ def bulk_add(self, works, operation, timestamp=None,
         # The SELECT part of the INSERT...SELECT query.
         new_records = _db.query(
             Work.id.label('work_id'), 
-            literal(operation, type_=BaseCoverageRecord.status_enum).label('operation'),
+            literal(operation, type_=String(255)).label('operation'),
```
Comment (leonardr) [inline model.py]: What's the motivation behind this change?

File: model.py [hunk 2]
```diff
@@ -1859,6 +1861,45 @@ def type_and_identifier_for_urn(cls, identifier_string):
         return (type, identifier_string)
 
     @classmethod
+    def parse_urns(cls, _db, identifier_strings):
```
Comment (leonardr) [inline model.py]: I see how this method saves time when you give it a bunch of identifiers that are already in the catalog. Is that all, or does it also save time when the identifiers are new?

File: model.py (hunk 1 above)
Comment (courte) [reply model.py]: I made this change when I was exploring bulk inserts. Type is supposed to be the target column type, and the column type of CoverageRecord.operation isn't an status enum, it's String.

If it's working just fine as-is, I can remove it.

Comment (leonardr) [reply model.py]: You're right, I mixed up operation and status originally. Thanks for fixing my error!

File: model.py (hunk 2 above)
Comment (courte) [reply model.py]: It doesn't save time when identifiers are new, but it could. Right now `Identifier.for_foreign_id` does some final evaluations on the identifier, but I'll refactor to apply them separately.

**Comment by leonardr** [comment]
Looking good!
